src/adapters/kafka/index.ts

```diff
@@ -0,0 +1,58 @@
+import { Kafka, Consumer, Producer } from 'kafkajs'
+import Adapter from '../../lib/adapter.js'
+import glee from '../../lib/glee.js'
+import GleeMessage from '../../lib/message.js'
+
+class KafkaAdapter extends Adapter {
+  private kafka: Kafka
+  private firstConnect: boolean = true
+  producer: any
+  name(): string {
+    return 'Kafka adapter'
+  }
+ 
+  async connect() {
+    const brokerUrl = new URL(this.AsyncAPIServer.url())
+    // kafka client
+    const kafka: Kafka = new Kafka({
+      clientId: 'glee-app',  // clientID: hardcoded need to change afterwards 
+      brokers: [brokerUrl.host],
+      ssl: true,
+      sasl: {
+        mechanism: 'plain',
+        username: process.env.KAFKA_USERNAME,
+        password: process.env.KAFKA_PASSWORD,
+      },
+    })
+
+    const consumer = kafka.consumer({ groupId: 'glee-group' })   // groupID: hardcoded need to change afterwards
+  
+    consumer.on('consumer.connect', () => {
+      if (this.firstConnect) {
+        this.firstConnect = false
+        this.emit('connect', {
+          name: this.name(),
+          adapter: this,
+          connection: consumer,
+          channels: this.channelNames
+        })
+      }
+    })
+    await consumer.connect() 
+    const subscribedChannels = this.getSubscribedChannels()
+    await consumer.subscribe({ topics: subscribedChannels, fromBeginning: true })
+    await consumer.run({
+      eachMessage: async ({ topic, partition, message }) => {
+        const prefix = `${topic}[${partition} | ${message.offset}] / ${message.timestamp}`
+        console.log(`- ${prefix} ${message.key}#${message.value}`)
```

Comment:
Here you have to create the GleeMessage and pass it over to Glee:

```suggestion
        const gleeMessage = this.createMessage(message)
        this.emit('message', gleeMessage)
```

Then you need to add another function called `createMessage` to this adapter. Similar to the one here: https://github.com/asyncapi/glee/blob/master/src/adapters/mqtt/index.ts#L151. The difference is that instead of getting an MQTT `packet` you get a Kafka `message`. You have to convert the message to a GleeMessage and return it. I think we discussed how to convert to GleeMessage on Slack.

Reply:
Actually, the way to convert a Kafka message to a GleeMessage is defined here: https://github.com/asyncapi/glee/pull/342#discussion_r1041781088.